java/vector/src/main/java/org/apache/arrow/vector/BaseVariableWidthViewVector.java

```diff
@@ -0,0 +1,1434 @@
+   * @param start start position of data in buffer
+   * @param end end position of data in buffer
+   * @param buffer data buffer containing the variable width element to be stored
+   *               in the vector
+   */
+  public void setSafe(int index, int isSet, int start, int end, ArrowBuf buffer) {
+    assert index >= 0;
+    final int dataLength = end - start;
+    handleSafe(index, dataLength);
+    fillHoles(index);
+    BitVectorHelper.setValidityBit(validityBuffer, index, isSet);
+    byte[] data = new byte[dataLength];
+    buffer.getBytes(start, data, 0, dataLength);
+    setBytes(index, data, 0, dataLength);
+    lastSet = index;
+  }
+
+  /**
+   * Store the given value at a particular position in the vector. isSet indicates
+   * whether the value is NULL or not.
+   * @param index position of the new value
+   * @param start start position of data in buffer
+   * @param length length of data in buffer
+   * @param buffer data buffer containing the variable width element to be stored
+   *               in the vector
+   */
+  public void set(int index, int start, int length, ArrowBuf buffer) {
+    assert index >= 0;
+    fillHoles(index);
+    BitVectorHelper.setBit(validityBuffer, index);
+    byte[] data = new byte[length];
+    buffer.getBytes(start, data, 0, length);
+    setBytes(index, data, start, length);
+    lastSet = index;
+  }
+
+  /**
+   * Same as {@link #set(int, int, int, int, ArrowBuf)} except that it handles the case
+   * when index is greater than or equal to current value capacity of the
+   * vector.
+   * @param index position of the new value
+   * @param start start position of data in buffer
+   * @param length length of data in buffer
+   * @param buffer data buffer containing the variable width element to be stored
+   *               in the vector
+   */
+  public void setSafe(int index, int start, int length, ArrowBuf buffer) {
+    assert index >= 0;
+    handleSafe(index, length);
+    fillHoles(index);
+    BitVectorHelper.setBit(validityBuffer, index);
+    byte[] data = new byte[length];
+    buffer.getBytes(start, data, 0, length);
+    setBytes(index, data, start, length);
+    lastSet = index;
+  }
+
+
+  /*----------------------------------------------------------------*
+   |                                                                |
+   |                helper methods for setters                      |
+   |                                                                |
+   *----------------------------------------------------------------*/
+
+
+  protected final void fillHoles(int index) {
+    for (int i = lastSet + 1; i < index; i++) {
+      setBytes(i, EMPTY_BYTE_ARRAY, 0, EMPTY_BYTE_ARRAY.length);
+    }
+    lastSet = index - 1;
+  }
+
+  /**
+   * Get the length of the view.
+   * @param index The index of the element in the vector.
+   * @return The length of the element at the given index.
+   */
+  protected final int getLength(int index) {
```

Comment:
```suggestion
  protected final int getValueLength(int index) {
```

Reply:
We can remove this method. I have introduced a duplicated functionality. Sorry about that.